### ISSUE TITLE
ether and wlan interfaces not progressing to routable configured state

### ISSUE THREAD

**Issue by ringlej** [systemd/systemd#21670]
**systemd version the issue has been seen with**
 > 249.2

**Used distribution**
 > Built from source for embedded system

**Linux kernel version used** (`uname -a`)

 > `Linux ec1c01g182100044 3.12.74-11-fs1.1 #2 2020-10-01T00:00:00+00:00 armv5tejl GNU/Linux`

**CPU architecture issue was seen on**
 > armv5te

**Expected behaviour you didn't see**
 > ether and wlan type interfaces don't go into `routeable configured` state

**Unexpected behaviour you saw**
 > ether and wlan type interfaces stay stuck in `carrier configuring` state

**Steps to reproduce the problem**
 > Wait for a couple minutes after bootup and check the output of `networkctl list`.
I performed a git bisect to find the systemd commit that introduced this issue:
~/git/systemd$ git bisect view --stat

commit 0828a38605975b68c14c9194a1ee2c5c2ff7038f (refs/bisect/bad)
Author: Yu Watanabe <watanabe.yu+github@gmail.com>
Date:   Mon Jun 7 21:53:35 2021 +0900

network: address: always read address flag from IFA_FLAGS attribute
    
Otherwise, update flag become incomplete and the IFA_F_MANAGETEMPADDR flag
will not be stored, thus no temporary addresses will be removed when
networkd requests to remove the main address.
    
Follow-up for a8481354f0cd2c0855472193d0f57c7a77674969.
Fixes #13218.
Fixes #19838.

 src/network/networkd-address.c | 23 ++++++++---------------
 1 file changed, 8 insertions(+), 15 deletions(-)

**Additional program output to the terminal or log subsystem illustrating the issue**

Expected output from systemd built on commit prior to the above mentioned bisect/bad commit:
```text
gpadmin@ec1c01g182100044:~$ networkctl list
IDX LINK  TYPE     OPERATIONAL SETUP     
  1 lo    loopback carrier     unmanaged
  2 net0  ether    routable    configured
  3 sit0  sit      off         unmanaged
  4 wlan0 wlan     routable    configured

4 links listed.
gpadmin@ec1c01g182100044:~$ networkctl status
●        State: routable                          
  Online state: online                            
       Address: 192.168.22.6 on net0
                172.22.22.22 on wlan0
                fe80::224:b7ff:fe00:11a9 on net0
                fe80::1ebf:ceff:fe27:43b0 on wlan0
       Gateway: 192.168.22.1 on net0
           DNS: 192.168.22.1

Dec 07 14:24:16 ec1c01g182100044 systemd-networkd[973]: wlan0: NDISC: Invoking callback for 'timeout' event.
Dec 07 14:24:16 ec1c01g182100044 systemd-networkd[973]: wlan0: NDisc handler get timeout event
Dec 07 14:24:16 ec1c01g182100044 systemd-networkd[973]: wlan0: link_check_ready(): dhcp4:no ipv4ll:no dhcp6_addresses:no dhcp_routes:no dhcp_pd_addresses:no dhcp_pd_routes:no ndisc_address>
Dec 07 14:24:16 ec1c01g182100044 systemd-networkd[973]: wlan0: State changed: configuring -> configured
Dec 07 14:24:16 ec1c01g182100044 systemd-networkd[973]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_34 interface=org.freedesktop.DBus.Propertie>
Dec 07 14:24:18 ec1c01g182100044 systemd-networkd[973]: Got message type=method_call sender=:1.15 destination=org.freedesktop.network1 path=/org/freedesktop/network1 interface=org.freedesk>
Dec 07 14:24:18 ec1c01g182100044 systemd-networkd[973]: Sent message type=method_return sender=n/a destination=:1.15 path=n/a interface=n/a member=n/a cookie=26 reply_cookie=2 signature=v >
Dec 07 14:24:18 ec1c01g182100044 systemd-networkd[973]: wlan0: NDISC: Sent Router Solicitation, next solicitation in 15s
Dec 07 14:24:21 ec1c01g182100044 systemd-networkd[973]: Got message type=method_call sender=:1.16 destination=org.freedesktop.network1 path=/org/freedesktop/network1 interface=org.freedesk>
Dec 07 14:24:21 ec1c01g182100044 systemd-networkd[973]: Sent message type=method_return sender=n/a destination=:1.16 path=n/a interface=n/a member=n/a cookie=27 reply_cookie=2 signature=v >
```
Unexpected behavior with systemd built on bisect/bad commit:
```text
gpadmin@ec1c01g182100044:~$ networkctl list
IDX LINK  TYPE     OPERATIONAL SETUP      
  1 lo    loopback carrier     unmanaged
  2 net0  ether    carrier     configuring
  3 sit0  sit      off         unmanaged
  4 wlan0 wlan     carrier     configuring

4 links listed.
gpadmin@ec1c01g182100044:~$ networkctl status
●        State: carrier
  Online state: offline
       Address: 192.168.22.6 on net0
                172.22.22.22 on wlan0
                fe80::224:b7ff:fe00:11a9 on net0
                fe80::1ebf:ceff:fe27:43b0 on wlan0
       Gateway: 192.168.22.1 on net0
           DNS: 192.168.22.1

Dec 07 14:46:24 ec1c01g182100044 systemd-networkd[973]: wlan0: link_check_ready(): an address 172.22.22.22/30 is not ready.
Dec 07 14:46:24 ec1c01g182100044 systemd-networkd[973]: Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/network1/link/_34 interface=org.freedesktop.DBus.Propertie>
Dec 07 14:46:24 ec1c01g182100044 systemd-networkd[973]: wlan0: Not emitting NTP servers on link, couldn't find suitable uplink.
Dec 07 14:46:24 ec1c01g182100044 systemd-networkd[973]: wlan0: Not emitting SIP servers on link, couldn't find suitable uplink.
Dec 07 14:46:24 ec1c01g182100044 systemd-networkd[973]: wlan0: DHCPv4 server: STARTED
Dec 07 14:46:24 ec1c01g182100044 systemd-networkd[973]: wlan0: Offering DHCPv4 leases
Dec 07 14:46:25 ec1c01g182100044 systemd-networkd[973]: wlan0: rtnl: received address message without flags, ignoring: No data available
Dec 07 14:46:25 ec1c01g182100044 systemd-networkd[973]: lo: Remembering foreign route: dst: fe80::1ebf:ceff:fe27:43b0/128, src: n/a, gw: n/a, prefsrc: n/a, scope: global, table: main(254),>
Dec 07 14:46:33 ec1c01g182100044 systemd-networkd[973]: Got message type=method_call sender=:1.8 destination=org.freedesktop.network1 path=/org/freedesktop/network1 interface=org.freedeskt>
Dec 07 14:46:33 ec1c01g182100044 systemd-networkd[973]: Sent message type=method_return sender=n/a destination=:1.8 path=n/a interface=n/a member=n/a cookie=13 reply_cookie=2 signature=v e>
Dec 07 14:46:37 ec1c01g182100044 systemd-networkd[973]: Got message type=method_call sender=:1.9 destination=org.freedesktop.network1 path=/org/freedesktop/network1 interface=org.freedeskt>
Dec 07 14:46:37 ec1c01g182100044 systemd-networkd[973]: Sent message type=method_return sender=n/a destination=:1.9 path=n/a interface=n/a member=n/a cookie=14 reply_cookie=2 signature=v e>
```

I believe the issue is happening because the flag IFA_F_TENTATIVE remains set on the interfaces and prevents it from going to the configured state:
> wlan0: link_check_ready(): an address 172.22.22.22/30 is not ready.

**Comment by ringlej** [comment]
Full systemd-networkd debug log here: https://gist.github.com/ringlej/713e652dad409877ba0d8f3a01b5b2ae

**Comment by yuwata** [comment]
> Linux ec1c01g182100044 3.12.74-11-fs1.1 #2 2020-10-01T00:00:00+00:00 armv5tejl GNU/Linux

Oh... Too old...

From our README:
```
REQUIREMENTS:
        Linux kernel >= 3.13
        Linux kernel >= 4.2 for unified cgroup hierarchy support
        Linux kernel >= 4.10 for cgroup-bpf egress and ingress hooks
        Linux kernel >= 4.15 for cgroup-bpf device hook
        Linux kernel >= 4.17 for cgroup-bpf socket address hooks
        Linux kernel >= 5.3 for bounded-loops in BPF program
        Linux kernel >= 5.4 for signed Verity images support
        Linux kernel >= 5.7 for BPF links and the BPF LSM hook
```

**Comment by yuwata** [comment]
Fix is waiting in #21672.

**Comment by ringlej** [comment]
Thanks! I can confirm that the #21672 fixes the issue
